architecture/workflows.md

```diff
@@ -0,0 +1,53 @@
+* Putting too much data on events will quickly re-introduce coupling and impact overall system performance. Read more about about properly sizing event messages in the blog post on [putting events on a diet](https://particular.net/blog/putting-your-events-on-a-diet).
+* Complex workflows that require aggregation of multiple events require local state. Service-internal [orchestration using NServiceBus sagas](#orchestration) can be used to easily aggregate multiple events and process timeouts in a component of a choreographed workflow.
+* Maintaining consistent state across all choreography participants in case of failures can become challenging as the coordination and flow of compensating events can quickly multiply the complexity of a choreographed workflow. Choreography is best used for processes that tolerate eventual consistency. [Orchestrated workflows](#orchestration) may better for processes that require more consistency.
+
+## Orchestration
+
+Orchestrated workflows are managed by a central process that instructs components to perform work in a specific order, manages state, and handles failures. Orchestration can be useful in complex workflows that contain multiple conditions or time-based triggers or have stronger consistency requirements. Instead of events, message-based orchestration relies on `commands` that the orchestrator sends to receivers which handle them.
+
+### Implementing orchestrated workflows
+
+NServiceBus is designed to handle long-running business processes in a robust and scalable way using the [saga feature](/nservicebus/sagas/). NServiceBus sagas are a convenient programming model to implement orchestrated, long-running business workflows or state machines using regular C# (or any other .NET language). Sagas handle recoverability, shared state, message correlation, timeouts, and more.
+
+The [ServiceInsight](/serviceinsight) saga view can show an orchestrated workflow implemented as an NServiceBus saga:
+
+![](/serviceinsight/images/overview-sagaview.png)
+
+[**Tutorial: Introduction to NServiceBus sagas →**](/tutorials/nservicebus-sagas/1-saga-basics/)
+
+Note: NServiceBus sagas focus on providing a convenient and efficient way to manage message-based workflows as described by the [Process Manager pattern](https://www.enterpriseintegrationpatterns.com/patterns/messaging/ProcessManager.html). The [_saga distributed transactions_ pattern](https://learn.microsoft.com/en-us/azure/architecture/reference-architectures/saga/saga) is primarily concerned with managing long-lived transactions and ensuring consistency between multiple operations, especially in the presence of failures. The NServiceBus saga feature can be used to implement the _saga distributed transactions_ pattern.
+
+### Challenges
```

Comment:
as in timeouts that arrive when the saga has already been completed?

Reply:
good point. I think those are valid challenges but they are also fairly technical and mostly connected to our implementation of saga timeouts? I think this might be something for the saga specific documentation rather than the concept itself? I'm on the fence but I could see this being eventually a bit too in-depth for readers of this page?